pkg/api/api.go

```diff
@@ -103,6 +103,26 @@ type IssueCodeResponse struct {
 	ErrorCode          string `json:"errorCode,omitempty"`
 }
 
+// CheckCodeStatusRequest defines the parameters to request the status for a
+// previously issued OTP code. This is called by the Web frontend.
+// API is served at /api/codestatus
+type CheckCodeStatusRequest struct {
+	ID []string `json:"id"`
+}
+
+// CheckCodeStatusResponse defines the response type for CheckCodeStatusRequest.
+type CheckCodeStatusResponse struct {
+	CodeStatus map[string]CodeStatus `json:"codeStatus"`
```

Comment:
What about for codes that are not found (i.e. have been garbage collected)?

Reply:
such codes will be omitted in the response